source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -53,47 +53,52 @@ void AMDSClientTCPSocket::onSocketError(QAbstractSocket::SocketError error)
 
 void AMDSClientTCPSocket::readClientRequestMessage()
```

Comment:
I've gone through this and I think the refactor works. It's certainly much cleaner code. I'm sure you've tested it for Continuous Request. Have you tested the other request types? The original decision flow was cumbersome, but it made it clear what each section was doing. Would you huck in some inline comments around the if/else blocks to inform about the decision flow?